File: InTravels.DAL/Entities/UserProfile.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace InTravels.DAL.Entities
{
    public class UserProfile
    {
        [Key]
        [ForeignKey("ApplicationUser")]
        public string Id { get; set; }

        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Address { get; set; }
        public int Age { get; set; }
        public Gender Gender { get; set; }
        public decimal Rating { get; set; }

        public virtual ICollection<UserProfile> Subscriptions { get; set; }
        public virtual ICollection<UserProfile> Subscribers { get; set; }
        public virtual ICollection<Post> Posts { get; set; }
        public virtual ICollection<Comment> Comments { get; set; }
        public virtual ApplicationUser ApplicationUser { get; set; }
    }

    public enum Gender : int
    {
        Male = 0, Female = 1
    }
}

File: Ang2/app/classes/userProfile.ts
export interface IUserProfile {
  username: string;
  password: string;
  Id: string;
  Email: string;
  FirstName: string;
  LastName: string;
  Gender: string;
  Age: number;
  Image: string;
}

export class UserProfile implements IUserProfile {
  username: string;
  password: string;
  Id: string;
  Email: string;
  FirstName: string;
  LastName: string;
  Gender: string;
  Age: number;
  Image: string;

  constructor() {
  }
}